From 4bcc93b8c01ca2b491d471c253ed75d8092a3705 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Guido=20G=C3=BCnther?= <agx@sigxcpu.org>
Date: Sat, 21 Nov 2009 21:29:52 +0100
Subject: [PATCH] multipath-toosl: Use current name of the divice node ($name)

instead of the kernel's name ($kernel). Otherwise we might end up
looking at a wrong or nonexistant node.
---
 kpartx/kpartx.rules |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/kpartx/kpartx.rules b/kpartx/kpartx.rules
index 5a62d57..8978b73 100644
--- a/kpartx/kpartx.rules
+++ b/kpartx/kpartx.rules
@@ -27,9 +27,9 @@ ENV{DM_PART}=="?*", \
 
 # Create dm tables for partitions
 ENV{DM_STATE}=="ACTIVE", ENV{DM_UUID}=="mpath-*", \
-        RUN+="/sbin/kpartx -a -p -part /dev/$kernel"
+        RUN+="/sbin/kpartx -a -p -part /dev/$name"
 ENV{DM_STATE}=="ACTIVE", ENV{DM_UUID}=="dmraid-*", \
-        RUN+="/sbin/kpartx -a -p -part /dev/$kernel"
+        RUN+="/sbin/kpartx -a -p -part /dev/$name"
 
 LABEL="kpartx_end"
 
-- 
1.7.0

